code/clue.js

```diff
@@ -1,59 +1,209 @@
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
+// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT PASSES IN TO THE FUNCTION.
 const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+	return array[Math.floor(Math.random() * array.length)]
+};
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+//OBJECT THAT KEEPS THE MYSTERY.
+const mystery = {
+	killer: "null",
+	weapon: "null",
+	room: "null"
+};
 
-// This function will be invoked when you click on the killer card.
+// KILLER CARD
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
-
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+	mystery.killer = randomSelector(suspects);
+	// This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer.
+	document.getElementById("killerCard").style.background = mystery.killer.color;
+	document.getElementById("killerName").innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+	document.getElementById("killerAge").innerHTML = `${mystery.killer.age} years old`;
+	document.getElementById("killerOccupation").innerHTML = mystery.killer.occupation;
+	document.getElementById("killerImage").src = mystery.killer.image;
+	document.getElementById("killerDescription").innerHTML = mystery.killer.description;
+};
+document.getElementById("killerCard").onclick = pickKiller;
+
+//WEAPON CARD
+const pickWeapon = () => {
+	mystery.weapon = randomSelector(weapons);
+	document.getElementById("weaponCard").style.background = mystery.weapon.color;
+	document.getElementById("weaponName").innerHTML = mystery.weapon.name;
+	document.getElementById("weaponWeight").innerHTML = `${mystery.weapon.weight} kg`;
+};
+document.getElementById("weaponCard").onclick = pickWeapon;
+
+//ROOM CARD
+const pickRoom = () => {
+	mystery.rooms = randomSelector(rooms);
+	document.getElementById("roomName").innerHTML = mystery.rooms;
+};
+document.getElementById("roomCard").onclick = pickRoom;
+
+//REVEAL THE MISTERY
+const theMystery = document.getElementById("mystery");
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+const revealMystery = () => {
+	theMystery.innerHTML = `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName} in the ${mystery.rooms} with a ${mystery.weapon.name}.`
```

Comment:
What happens if you click reveal without clicking any cards?  Is there a way to make the message clearer for the user?